pkg/api/helpers/upload.go

```diff
@@ -6,21 +6,75 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"mime"
+	"mime/multipart"
 	"net/http"
+	"net/textproto"
 	"net/url"
+	"path/filepath"
 	"strings"
 
 	"github.com/go-openapi/swag"
 	"github.com/treeverse/lakefs/pkg/api"
 )
 
+// ClientUpload uploads contents as a file via lakeFS
+func ClientUpload(ctx context.Context, client api.ClientWithResponsesInterface, repoID, branchID, objPath string, metadata map[string]string, contentType string, contents io.ReadSeeker) (*api.ObjectStats, error) {
+	pr, pw := io.Pipe()
+	defer func() {
+		_ = pr.Close()
+	}()
+	mpw := multipart.NewWriter(pw)
+	mpContentType := mpw.FormDataContentType()
+	go func() {
+		defer func() {
+			_ = pw.Close()
```

Comment:
1. We should close `mpw` before `pw` - consider moving `mpw.Close` here
2. Verify if eveything works while calling `pw.CloseWithError` followed by all the defer Close class. (double close)

Reply:
2. Works